modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/source-builder/serializeUtil.js

```diff
@@ -92,7 +92,7 @@ function appendXMLActions(
 				);
 			}
 
-			if (isValidValue(status, index)) {
+			if (isValidValue(status, index) && status[index]) {
```

Comment:
THe definition of `isValidValue` is as follows:
```
function isValidValue(array, index) {
	return array && array[index] !== undefined;
}
```
When the problem occurs, do we have a `null` at index?

Try changing the definition of `isValidValue` to

```
function isValidValue(array, index) {
	return array && array[index] !== undefined && array[index] !== null;
}
```
And see if it fixes.

Reply:
I am suggesting this because `isValidValue` already checks if `status[index]` is `undefined`, so it takes values as `0`, empty string or null as valid. What is not valid in this context? What is the value of `status[index]` that is causing the problem and that we should account for?

Reply:
As we have discussed in person, we will keep the solution as it is now.